dags/dag_content/tower_metrics_functions.py

```diff
@@ -311,16 +277,16 @@ def update_secret(
         "password": password,
     }
     secret_string = json.dumps(secret_dict)
-    response = secrets.update_secret(
+    response = secrets_client.update_secret(
         SecretId="Programmatic-DB-Clone-Access", SecretString=secret_string
     )
     secret_arn = response["ARN"]
-
+    time.sleep(30) #query_database always fails the first time because secret modification isnt done yet
```

Comment:
This is here because after refactoring, I noticed that the `query_database` task always fails on the first try. Since I know that the db status is "available" thanks to the waiter at the end of `modify_database`, I know that it has to have something to do with the modification of the secret. There is no waiter for mortifying secrets, and my attempts to create a waiter failed given that the secret does seem to be updated (new password applied) right away. It works consistently with this 30-second sleep at the moment, but I know that this is not ideal since it doesn't have anything concrete to check. Any suggestions?